calyx-backend/src/verilog.rs

```diff
@@ -539,8 +539,8 @@ fn emit_assignment(
 
     // If this is a data port
     let rhs: v::Expr = if is_data_port(dst) {
-        if assignments.len() == 1 {
```

Comment:
Could you factor the Verilog backend changes into a separate PR? I am not sure why we would need this.
It might be possible to just mark the `cell` that you create in your instrumentation pass with `@control` which would be a cleaner solution.

Reply:
Good catch. Instrumentation should not change the backend!

Reply:
I'll try looking into this! I think the idea here is that we get the instrumentation pass to override `std_wire`'s `@data` input?

Reply:
Can you describe the interaction between `@data` and `@protected`? Why is something being marked as `@protected` change its data semantics?

Reply:
Marking the cell as `@control` fixed the problem so I removed the Verilog backend changes!

Reply:
So the instrumentation generates protected control signal? What is the rationale for these signals being control?

I ask because if the semantics of control change in the future, it will silently break instrumentation if we dont have a precise reasoning for why these signals are marked control.

Reply:
That's fair - the problem is that I'm trying to use `std_wire` for instrumented cells, but I can't have the Verilog backend reduce the assignment to `<group>_probe_in = 1'd1` (which it will if I don't make the cells `@control`, because the input port of `std_wire` is `@data`).

I think there are a couple of paths that we can take, most of which I've explored so far: (1) have a group that initializes all `<group>_probe.in` to `0` (which is not ideal because we'd need extra cycles to execute this group first) (2) make a new primitive that doesn't have a `@data` input port, (3) have the instrumentation pass make its cells `@control` (which is my current changes here), and (4) have `@protected` mean that the Verilog backend should preserve the original assignment (my original changes here).
If you have other suggestions though, I'd be happy to hear them!

Reply:
Can you explain how the probes work? It seems you want them to be set to `1` when a particular group is executing and otherwise have them be `0`? If so, the `@control` annotation makes sense to me since it provides this precise guarantee: when there is an explicit assignment, it takes that value; otherwise, it asserts `0`.

Reply:
Yup that's exactly how the probes work!! I'm glad to hear that makes sense - I'll write some comments documenting the reason why we want it to be a `@control` cell.